internal/common/predicate.go

```diff
@@ -12,3 +14,14 @@ func IsControlledBy(object client.Object, owner *metav1.OwnerReference) bool {
 		controller.APIVersion == owner.APIVersion &&
 		controller.Kind == owner.Kind
 }
+
+// FetchCurrentNamespaceName returns namespace name by using information stored as file
+// Returns default Openshift Builds namespace on error
+// Refer: https://kubernetes.io/docs/tasks/run-application/access-api-from-pod/#without-using-a-proxy
+func FetchCurrentNamespaceName() string {
+	content, err := os.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace")
+	if err != nil {
+		return "openshift-builds"
+	}
+	return string(content)
```

Comment:
Do we need to worry about line endings or other invalid characters?

Reply:
No, because the file has only one string with no line ending, also we are doing a file read instead of line read and then passing the value to a byte to string converter so that ensure that our string is free of other chars.